### PR TITLE
Code split select heavy views (stream, issue details, etc)

### PR DESCRIPTION
**Pull request by benvinegar**
This patch code splits/lazy loads the following views/routes:

* app/views/stream.jsx
* app/views/groupEventDetails.jsx
* app/views/groupSharedDetails.jsx
* app/views/groupActivity.jsx

This saves around **~700kb unminified (76 KB after minfiication + gzip)** from app.js, because these views are either themselves heavy (stream.jsx, groupEventDetails.jsx) or have heavy dependencies like react-mentions (groupActivity.jsx).

| Bundled file | Before | Before (min + gzip) | After | After (min + gzip) |
| --- | --- | --- | --- | --- |
| app.js | 1.83 MB | 215 KB | 1.13 MB | 139 KB |
| 0.js (groupEventDetails) | --- | ---| 200 KB | 26 KB |
| 1.js (stream) | --- | --- | 115 KB | 15 KB |
| 2.js (groupActivity) | --- | --- | 400 KB | 37 KB |
| 3.js (groupSharedDetails) | --- | --- | 149 KB | 19 KB |

Before:

image

After:

image

**NOTE**: Should definitely verify this build on sentry.io staging before deploying.

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">You need to update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/routes.jsx [hunk 1]
```diff
@@ -207,8 +212,17 @@ function routes() {
         </Route>
 
         <Route path=":projectId/" component={errorHandler(ProjectDetails)}>
-          <IndexRoute component={errorHandler(Stream)} />
-          <Route path="searches/:searchId/" component={errorHandler(Stream)} />
+          <IndexRoute
```
Comment (dcramer) [inline src/sentry/static/sentry/app/routes.jsx]: why not make this into like LazyRoute/LazyIndexRoute

Comment (benvinegar) [inline src/sentry/static/sentry/app/routes.jsx]: I mean, sure, afterwards.

**Comment by benvinegar** [comment]
If you're wondering about that big lodash blob in 2.js, see #5616

**Comment by dcramer** [comment]
I'd really like to see us split views not as one-offs, but by system. e.g. project/org settings should be broken off, /manage/ should be broken off. Stream is fine here, but in reality most people are loading sentry for either the stream or the issue details pages and while theres a fair argument to be made they might not hit the stream, it seems like we're just saying "this page is big, lets make it its own js file"

**Comment by benvinegar** [comment]
> I'd really like to see us split views not as one-offs, but by system. e.g. project/org settings should be broken off, /manage/ should be broken off. Stream is fine here, but in reality most people are loading sentry for either the stream or the issue details pages and while theres a fair argument to be made they might not hit the stream, it seems like we're just saying "this page is big, lets make it its own js file"

I definitely want to do follow-ups. This is just a nice first step, and also illustrates how this is possible.

Even if they hit stream + group details first, this basically feels the same as it does before. But if they hit one of the settings pages (which loads all of vendor.js and app.js), or sign in and land on the org dashboard, it will feel much faster.

**Review by dcramer — APPROVED** [review]
im good with this once we confirm its not going to break the world

**Comment by billyvg** [comment]
Closing in favor of https://github.com/getsentry/sentry/pull/7219 (does not split stream though)

**Pull request closed without merging**